### PR TITLE
Add org.devsk.flatconvert

### PR DESCRIPTION
**Pull request by quentinSoranzoKrebs**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.  
**FlatConvert** is a simple and user-friendly multimedia conversion tool for GNOME. It allows users to convert video and image files to various formats using FFmpeg and Python Pillow under the hood. It integrates well with GNOME by following modern design patterns with GTK4 and libadwaita.

- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.  
Application ID: `org.devsk.flatconvert` — The domain `devsk.org` (and the `qsk` namespace on GNOME GitLab) is controlled by the developer.

- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.

- [x] I am an _author/developer_ of the project.

Additional maintainers: *None*

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository  
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id  
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements  
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.devsk.flatconvert.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    "runtime-version" : "48",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flatconvert",
+	"finish-args": [
+	  "--share=network",
+	  "--share=ipc",
+	  "--socket=fallback-x11",
+	  "--filesystem=home",
+	  "--device=dri",
+	  "--socket=x11",
```
Comment (hfiguiere) [inline org.devsk.flatconvert.json:13]: it either `wayland` + `fallback-x11` or just `x11`. Not both.

**Comment by hfiguiere** [comment]
> * The domain `devsk.org`

There is nothing responding at this end.

File: org.devsk.flatconvert.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://gitlab.gnome.org/qsk/flatconvert/-/archive/1.1.0/flatconvert-1.1.0.tar.gz",
+		    "sha256" : "aaeb0d9ea198fd423bdf6fb554a041ad63ddd5b776f02abc43c83c1e71120438"
+		}
+            ],
+            "config-opts" : [
+                "--libdir=lib"
```
Comment (hfiguiere) [inline org.devsk.flatconvert.json:101]:
```suggestion
```

this is unecessary.

File: icons/org.devsk.flatconvert.svg
Comment (hfiguiere) [inline icons/org.devsk.flatconvert.svg:1]: once again. this file a never referenced nor used.

File: metainfo/org.devsk.flatconvert.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component
+    xmlns="http://www.freedesktop.org/standards/appstream"
+    xmlns:its="http://www.w3.org/2005/11/its"
+    type="desktop">
+  <id>org.devsk.flatconvert.desktop</id>
```
Comment (hfiguiere) [inline metainfo/org.devsk.flatconvert.metainfo.xml:6]: id never ends with `.desktop`

File: metainfo/org.devsk.flatconvert.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  <name its:translate="yes">flatconvert</name>
+  <summary its:translate="yes">A file conversion application for the gnome desktop environnement.</summary>
+  <url type="homepage">https://gitlab.gnome.org/qsk/flatconvert</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-or-later</project_license>
+  <description its:translate="yes">
+    <p>A file conversion application for the gnome desktop environnement.</p>
+  </description>
+</component>
```
Comment (hfiguiere) [inline metainfo/org.devsk.flatconvert.metainfo.xml:15]: missing required `url`, `content_rating`, `releases`, `screenshots`.

Also `description` should not be a repeat of `summary`.

**Comment by quentinSoranzoKrebs** [comment]
> > * The domain `devsk.org`
> 
> There is nothing responding at this end.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
You are far from having resolved all the issues.

**Comment by hfiguiere** [comment]
> Build must be offline.

This was never address. Fix ALL the issue before asking for review.

File: icons/org.devsk.flatconvert.svg
Comment (hfiguiere) [reply icons/org.devsk.flatconvert.svg:1]: this was NEVER addressed.

File: metainfo/org.devsk.flatconvert.metainfo.xml
[the remark this answers is not in the document]
Comment (hfiguiere) [reply metainfo/org.devsk.flatconvert.metainfo.xml:1]: this was NEVER addressed.

File: org.devsk.flatconvert.svg
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.devsk.flatconvert.svg:1]: this was NEVER addressed.

File: screenshots/screenshot1.png
Comment (hfiguiere) [inline screenshots/screenshot1.png:1]: why are screenshots here? This was never requested.

File: org.devsk.flatconvert.json [hunk 5]
```diff
@@ -0,0 +1,103 @@
+{
+    "app-id" : "org.devsk.flatconvert",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "48",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flatconvert",
+    "metadata": {
+    	"file": "metainfo/org.devsk.flatconvert.metainfo.xml"
+    },
```
Comment (hfiguiere) [inline org.devsk.flatconvert.json:9]: what documentation said this was valid? IT IS INVALID.

File: org.devsk.flatconvert.json [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    "finish-args": [
+	"--share=network",
+	"--share=ipc",
+	"--filesystem=home",
+	"--device=dri",
+	"--socket=wayland",
```
Comment (hfiguiere) [inline org.devsk.flatconvert.json:15]: what happen on an X11 session?

File: screenshots/screenshot1.png
Comment (quentinSoranzoKrebs) [reply screenshots/screenshot1.png:1]: ✅ Fixed. I’ve removed the local screenshot files from the repository.
Screenshots are now hosted externally as recommended:
https://devsk.org/screenshots/screenshot3.png
Thanks for the feedback!

**Comment by quentinSoranzoKrebs** [comment]
> > * The domain `devsk.org`
> 
> There is nothing responding at this end.

**Comment by quentinSoranzoKrebs** [comment]
I have addressed all the points raised: the offline build has been successfully verified, the devsk.org domain is now active, and the files have been thoroughly cleaned.
I would appreciate it if you could kindly review the pull request again at your earliest convenience.

Thank you very much for your time and consideration.

File: org.devsk.flatconvert.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        "--socket=fallback-x11",
+        "--filesystem=home",
+        "--device=dri",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.FileManager1",
+        "--env=FLATPAK_ENABLE_FFMPEG=1"
+    ],
+    "add-extensions": {
+        "org.freedesktop.Platform.ffmpeg": {
+            "version": "23.08",
```
Comment (hfiguiere) [inline org.devsk.flatconvert.json:19]: this version is incorrect. GNOME 48 is based on 24.08. And please indent things properly. (line below)

File: org.devsk.flatconvert.json [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--filesystem=home",
+        "--device=dri",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.FileManager1",
+        "--env=FLATPAK_ENABLE_FFMPEG=1"
+    ],
+    "add-extensions": {
+        "org.freedesktop.Platform.ffmpeg": {
```
Comment (bbhtt) [inline org.devsk.flatconvert.json]: you need add-ld-path too see the example here https://docs.flatpak.org/en/latest/extension.html#loading-existing-extensions

Comment (bbhtt) [reply org.devsk.flatconvert.json]: and create the directory

File: org.devsk.flatconvert.json (hunk 7 above)
Comment (quentinSoranzoKrebs) [reply org.devsk.flatconvert.json]: Thank you for the feedback.
I have changed the extension version to `24.08` to match the runtime version, and updated the JSON indentation to 4 spaces.
All changes are included in commit `805bd8f`.
I didn’t notice any other indentation issues, but feel free to point them out if I missed something.

File: org.devsk.flatconvert.json (hunk 8 above)
Comment (quentinSoranzoKrebs) [reply org.devsk.flatconvert.json]: Thank you for the suggestions.
I have added `add-ld-path` and ensured the required directory is created as per the Flatpak documentation.
All changes are included in commit `805bd8f`.
Please let me know if anything else needs adjustment.

**Comment by quentinSoranzoKrebs** [comment]
Hello,
I have implemented the requested changes, including adding `add-ld-path`, specifying the creation of the required directory both in the Flatpak manifest and ensuring it exists in the sandboxed environment of the app, and updating the extension version to `24.08` (see commit `805bd8f`).
Could you please have another look when you have time?
Thank you very much!

File: org.devsk.flatconvert.json [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    "app-id": "org.devsk.flatconvert",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "48",
+    "sdk": "org.gnome.Sdk",
+    "command": "flatconvert",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--filesystem=home",
```
Comment (bbhtt) [inline org.devsk.flatconvert.json]: why is home needed? does this not support filechooser portal?

Comment (quentinSoranzoKrebs) [reply org.devsk.flatconvert.json]: Thank you for your comment. Access to home is no longer necessary as I have implemented support for the filechooser portal. This change was made in commit e6da701, where I removed access to home and used only the filemanager portal to resolve this issue. This should address your concerns.

**Comment by quentinSoranzoKrebs** [comment]
bot, build

File: org.devsk.flatconvert.json [hunk 10]
```diff
@@ -55,7 +55,7 @@
                 {
                     "type": "archive",
                     "url": "https://gitlab.gnome.org/qsk/flatconvert/-/archive/v1.2.0/flatconvert-v1.2.0.tar.gz",
-                    "sha256": "c30674532e66ba357f2abb8910f9013327485e98009b5bedd3559df40b44929e"
+                    "sha256": "3ff408480c804a9e184d178c36ca9f3e47c0d5bb78c370d9463a5bb27286bf88"
```
Comment (hfiguiere) [inline org.devsk.flatconvert.json]: retagging is bad.

Comment (quentinSoranzoKrebs) [reply org.devsk.flatconvert.json]: Thank you for your comment. I am currently making updates to my GNOME GitLab repository, which caused the SHA-256 inconsistency. I have fixed this issue in commit 0d60b92.

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
why do these start with `_`?

image

missing icon

image

missing icon + why "test page"? preferences are empty

image

what is "Aide" here? it opens the same dialogue as above

image

after conversion where does the file get saved? I see `20:05:24,091 - INFO - convert_thread.py - utilisation de _convert_img pour /run/user/1000/doc/d749b1fd/file_example_JPG_2500kB_converted.png` in the terminal but i can't find it.

File: org.devsk.flatconvert.json [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    "sdk": "org.gnome.Sdk",
+    "command": "flatconvert",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.FileManager1",
+        "--env=FLATPAK_ENABLE_FFMPEG=1"
```
Comment (bbhtt) [inline org.devsk.flatconvert.json]: why do you need an env var and for what? just use FFmpeg if it exists.

File: org.devsk.flatconvert.json [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    "modules": [
+        "python3-requirements.json",
+        {
+            "name": "libmagic",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://astron.com/pub/file/file-5.45.tar.gz",
+                    "sha256": "fc97f51029bb0e2c9f4e3bffefdaf678f0e039ee872b9de5c002a6d09c784d82"
```
Comment (bbhtt) [inline org.devsk.flatconvert.json]: libmagic is in the runtime:

```
flatpak run --command=find org.gnome.Platform//48 /usr -name "*libmagic*"
/usr/lib/x86_64-linux-gnu/libmagic.so.1
/usr/lib/x86_64-linux-gnu/libmagic.so.1.0.0
```

File: org.devsk.flatconvert.json [hunk 13]
```diff
@@ -0,0 +1,63 @@
+{
+    "app-id": "org.devsk.flatconvert",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "48",
+    "sdk": "org.gnome.Sdk",
+    "command": "flatconvert",
+    "finish-args": [
+        "--share=network",
```
Comment (bbhtt) [inline org.devsk.flatconvert.json]: i can't see a reason why it needs network.

**Comment by bbhtt** [comment]
bot, build
